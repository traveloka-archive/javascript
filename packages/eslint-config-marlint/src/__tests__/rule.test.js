const fs = require('fs');
const path = require('path');
const eslint = require('eslint');
const tempWrite = require('temp-write');

const DEFAULT_CONFIG = require('../../index');

function runEslint(file, conf = DEFAULT_CONFIG) {
  const str = fs.readFileSync(file, { encoding: 'utf-8' });
  const linter = new eslint.CLIEngine({
    useEslintrc: false,
    configFile: tempWrite.sync(JSON.stringify(conf)),
  });

  return linter
    .executeOnText(str)
    .results[0].messages // disable marlint plugin for now because eslint failed to load it
    .filter(m => !m.ruleId.startsWith('marlint'))
    .sort((m1, m2) => {
      // sort by line number first
      if (m1.line > m2.line) {
        return 1;
      }

      if (m1.line < m2.line) {
        return -1;
      }

      // same line number, sort by column
      if (m1.column > m2.column) {
        return 1;
      }

      if (m1.column < m2.column) {
        return -1;
      }

      // same line number, same column, sort by rule id
      if (m1.ruleId > m2.ruleId) {
        return 1;
      }

      return -1;
    });
}

it('block offending rule', async () => {
  const file = path.join(__dirname, './fixture-error.js');
  let errors = await runEslint(file);

  errors = errors.map(err => ({
    ruleId: err.ruleId,
    severity: err.severity,
    line: err.line,
    column: err.column,
  }));

  expect(errors).toMatchSnapshot();
});

it('pass on correct code', async () => {
  const file = path.join(__dirname, './fixture-pass.js');
  const errors = await runEslint(file);

  expect(errors.length).toEqual(0);
});
