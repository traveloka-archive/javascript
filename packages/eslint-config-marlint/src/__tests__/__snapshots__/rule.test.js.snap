// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`block offending rule 1`] = `
Array [
  Object {
    "column": 1,
    "line": 7,
    "ruleId": "lodash/import-scope",
    "severity": 2,
  },
  Object {
    "column": 10,
    "line": 7,
    "ruleId": "no-unused-vars",
    "severity": 2,
  },
  Object {
    "column": 3,
    "line": 13,
    "ruleId": "react/no-unused-prop-types",
    "severity": 1,
  },
  Object {
    "column": 3,
    "line": 20,
    "ruleId": "react/no-unused-prop-types",
    "severity": 1,
  },
  Object {
    "column": 16,
    "line": 24,
    "ruleId": "flowtype/require-parameter-type",
    "severity": 1,
  },
  Object {
    "column": 10,
    "line": 26,
    "ruleId": "react/react-in-jsx-scope",
    "severity": 2,
  },
  Object {
    "column": 7,
    "line": 32,
    "ruleId": "no-unused-vars",
    "severity": 2,
  },
  Object {
    "column": 10,
    "line": 36,
    "ruleId": "react/react-in-jsx-scope",
    "severity": 2,
  },
  Object {
    "column": 33,
    "line": 36,
    "ruleId": "react/prop-types",
    "severity": 1,
  },
  Object {
    "column": 21,
    "line": 41,
    "ruleId": "no-undef",
    "severity": 2,
  },
  Object {
    "column": 14,
    "line": 47,
    "ruleId": "flowtype/require-parameter-type",
    "severity": 1,
  },
  Object {
    "column": 20,
    "line": 47,
    "ruleId": "no-empty-function",
    "severity": 2,
  },
  Object {
    "column": 7,
    "line": 56,
    "ruleId": "react/react-in-jsx-scope",
    "severity": 2,
  },
  Object {
    "column": 1,
    "line": 64,
    "ruleId": "no-var",
    "severity": 2,
  },
  Object {
    "column": 5,
    "line": 68,
    "ruleId": "prefer-const",
    "severity": 1,
  },
  Object {
    "column": 3,
    "line": 71,
    "ruleId": "object-shorthand",
    "severity": 1,
  },
  Object {
    "column": 3,
    "line": 79,
    "ruleId": "func-names",
    "severity": 2,
  },
  Object {
    "column": 3,
    "line": 79,
    "ruleId": "prefer-arrow-callback",
    "severity": 2,
  },
  Object {
    "column": 12,
    "line": 81,
    "ruleId": "react/react-in-jsx-scope",
    "severity": 2,
  },
  Object {
    "column": 3,
    "line": 83,
    "ruleId": "flowtype/require-parameter-type",
    "severity": 1,
  },
  Object {
    "column": 10,
    "line": 89,
    "ruleId": "flowtype/require-parameter-type",
    "severity": 1,
  },
  Object {
    "column": 15,
    "line": 89,
    "ruleId": "flowtype/require-parameter-type",
    "severity": 1,
  },
  Object {
    "column": 1,
    "line": 98,
    "ruleId": "no-console",
    "severity": 1,
  },
  Object {
    "column": 10,
    "line": 100,
    "ruleId": "no-unused-vars",
    "severity": 2,
  },
  Object {
    "column": 14,
    "line": 100,
    "ruleId": "flowtype/require-parameter-type",
    "severity": 1,
  },
  Object {
    "column": 5,
    "line": 103,
    "ruleId": "react-hooks/rules-of-hooks",
    "severity": 2,
  },
  Object {
    "column": 35,
    "line": 107,
    "ruleId": "react-hooks/exhaustive-deps",
    "severity": 1,
  },
  Object {
    "column": 10,
    "line": 111,
    "ruleId": "no-unused-vars",
    "severity": 2,
  },
  Object {
    "column": 5,
    "line": 113,
    "ruleId": "react/react-in-jsx-scope",
    "severity": 2,
  },
  Object {
    "column": 7,
    "line": 114,
    "ruleId": "react/react-in-jsx-scope",
    "severity": 2,
  },
  Object {
    "column": 3,
    "line": 123,
    "ruleId": "react-native/no-unused-styles",
    "severity": 2,
  },
]
`;
